### PR TITLE
#182210066 Users should be able to chat on Barefoot Nomad

### PR DESCRIPTION
**Pull request by nyakamwe**
**What does this PR do?**
It returns all public chats to the users.

**Description of Task to be completed?**
    Create a Chat model
    Associates(relates) each user to the corresponding chats

**How should this be manually tested?**
After cloning this repo:
1. Run `npm install` for installing all dependencies.
2. Start the development server by running `npm run dev`
3. Use testing tools like postman or swagger to test this endpoint `api/users/chats`

**What are the relevant pivotal tracker stories?**
[ #182210066](https://www.pivotaltracker.com/n/projects/2569261/stories/182210066)

### CODE REVIEW

File: src/controllers/chatControllers.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import chatService from "../services/chatServices";
+
+class ChatController{
+    static async getAllChatMessages(req,res){
+        const messages = await chatService.retrieveMessages();
+
+        return res.status(200).json({status:200, count:messages.length, chats:messages});
```
Comment (habaumugisha1) [inline src/controllers/chatControllers.js:7]: what is count? @nyakamwe

File: src/database/migrations/20220625143812-create-chat.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  async up(queryInterface, Sequelize) {
+    await queryInterface.createTable('chats', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      sender: {
+        allowNull: false,
```
Comment (habaumugisha1) [inline src/database/migrations/20220625143812-create-chat.js:12]: what are the differences between sender and postedBy

File: src/controllers/chatControllers.js (hunk 1 above)
Comment (nyakamwe) [reply src/controllers/chatControllers.js:7]: this count refer to  the number of messages or chats that are in db.

File: src/database/migrations/20220625143812-create-chat.js (hunk 2 above)
Comment (nyakamwe) [reply src/database/migrations/20220625143812-create-chat.js:12]: sender holds string while postedBy holds id

**Review by istafrica-dennw — APPROVED** [review]

**Pull request merged**